generate.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import datetime
+import pytz
+import math
+import numpy as np
+import timely_beliefs as tb
+from datetime import timedelta
+from timely_beliefs.beliefs.utils import load_time_series
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+def get_observation(csv_in,current_time):
+    """
+    Returns observation at given datetime
+
+    @param csv_in : csv file containing all forecast data
+    @param current_time : datetime string to observe
+    @param model : model to use to generate new data
+    """
+    with open(csv_in) as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        data = list(csv_reader)[1:]
+    index = get_row(current_time,data)
+    return float(data[index][1])
+
+def generator(csv_in, current_time, model):
+    """
+    Generates a new forecast at a given time using a model
+
+    @param csv_in : csv file containing all forecast data
+    @param current_time : datetime string to observe
+    @param model : model to use to generate new data
+    """
+    observation = get_observation(csv_in, current_time)
+    if str(type(model)) == "<class 'sklearn.linear_model.base.LinearRegression'>":
+        X = np.array([[1], [2], [3], [4]]) # placeholder
+        model.fit(X,X)
+        return model.predict(np.array([[observation]]))[0][0]
+    print("Other model functionality has not yet been implemented")
+    return observation
+
+#input
+#output rows of timely belief structure as array
+def main(csv_in,current_time,start_time,last_start_time,model=LinearRegression(), addtocsv = False):
+    with open(csv_in) as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        datacomp = list(csv_reader)
+        data = datacomp[1:]
+    index_current = get_row(current_time,data)
+    index_start = get_row(start_time,data)
+    index_last_start = get_row(last_start_time,data)
+    with open('temp.csv', 'w') as csvfile0:
+        writer = csv.writer(csvfile0, delimiter=',')
+        index_list = range(index_start,index_last_start+1)
+        result = []
+        for index in index_list:
+            value = generator(csv_in, current_time, model)
+            result += [[data[index][0],data[index_current][0],'Test',0.5,value]]
+            if addtocsv == True:
+                # print((index-index_current)*0.15)
+                datacomp[index][round((index-index_current)*0.15)] = value
+        columns = ["event_start","belief_time","source","cumulative_probability","event_value"]
+        writer.writerow(columns)
+        writer.writerows(result)
+        # print(datacomp[index][round((index-index_current)*0.15)])
+    t = pd.read_csv("temp.csv",index_col=0)
+    if addtocsv == True:
+        with open(csv_in, 'w') as csvfile:
+            writer = csv.writer(csvfile, delimiter=',')
+            writer.writerows(datacomp)
+    return t
```

Comment:
Function returns a DataFrame that looks like a BeliefsDataFrame, but is not actually a BeliefsDataFrame.